tests/suites/test_suite_pk.function

```diff
@@ -1909,56 +1928,65 @@ void pk_psa_sign(int curve_or_keybits, int psa_type, int expected_bits)
                                       hash, sizeof(hash), sig, sig_len) == 0);
     }
 
-    /* Export underlying public key for re-importing in a psa context. */
+    /* Export public key from the opaque PK context. */
 #if defined(MBEDTLS_PK_WRITE_C)
-    ret = mbedtls_pk_write_pubkey_der(&pk, pkey_psa,
-                                      sizeof(pkey_psa));
+    ret = mbedtls_pk_write_pubkey_der(&pk, opaque_pub_key, sizeof(opaque_pub_key));
     TEST_ASSERT(ret >= 0);
-    klen_psa = (size_t) ret;
+    opaque_pub_key_len = (size_t) ret;
     /* mbedtls_pk_write_pubkey_der() writes backwards in the data buffer. */
-    pkey_psa_start = pkey_psa + sizeof(pkey_psa) - klen_psa;
-#else
-    psa_status_t status;
-
-    status = psa_export_public_key(key_id, pkey_psa, sizeof(pkey_psa),
-                                   &klen_psa);
-    TEST_EQUAL(status, PSA_SUCCESS);
-    pkey_psa_start = pkey_psa;
+    memmove(opaque_pub_key,
+            opaque_pub_key + sizeof(opaque_pub_key) - opaque_pub_key_len,
+            opaque_pub_key_len);
+#else /* MBEDTLS_PK_WRITE_C */
+    TEST_EQUAL(psa_export_public_key(key_id, opaque_pub_key, sizeof(opaque_pub_key),
+                                     &opaque_pub_key_len), PSA_SUCCESS);
 #endif /* MBEDTLS_PK_WRITE_C */
 
-    TEST_ASSERT(klen_psa == klen_legacy);
-    TEST_ASSERT(memcmp(pkey_psa_start, pkey_legacy_start, klen_psa) == 0);
+    /* Check that the public keys of opaque and non-opaque PK contexts match. */
+    TEST_EQUAL(opaque_pub_key_len, legacy_pub_key_len);
+    TEST_MEMORY_COMPARE(opaque_pub_key, opaque_pub_key_len, legacy_pub_key, legacy_pub_key_len);
 
+    /* Destroy the opaque PK context and the wrapped PSA key. */
     mbedtls_pk_free(&pk);
     TEST_ASSERT(PSA_SUCCESS == psa_destroy_key(key_id));
 
-    mbedtls_pk_init(&pk);
-
-    /* If we used "pk_write" previously, then we go for a "pk_parse" here;
+    /* Create a new non-opaque PK context to verify the signature.
+     *
+     * Note: if we used "pk_write" previously, then we go for a "pk_parse" here;
```

Comment:
Mmmm right, this guard doesn't look correct at all. We should only check `MBEDTLS_PK_PARSE_C` here since we're only going to parse the key not writing it.

Reply:
The guards need to be the same because the export formats are different (with or without metadata). I think that's why the CI is failing on ea01efa589cde8e984f37a9aa1898825532b3ec6 in a configuration with pkparse but not pkwrite (`configs/config-suite-b.h`). I guess we don't have a configuration with pkwrite but not pkparse, so the CI could pass on 1b533ab205f4674dba31b563a357db464f96a39f.